stubs/requests/requests/sessions.pyi

```diff
@@ -113,7 +113,9 @@ class _Settings(TypedDict):
 
 class Session(SessionRedirectMixin):
     __attrs__: Any
-    headers: CaseInsensitiveDict[str | bytes]
+    # See https://github.com/psf/requests/issues/5020#issuecomment-989082461:
+    # requests sets this as a CaseInsensitiveDict, but users may set it to any MutableMapping
+    headers: MutableMapping[str, str | bytes | None]
```

Comment:
Are you sure this should be `MutableMapping[str, str | bytes | None]` rather than `MutableMapping[str, str | bytes]`? This annotation was debated somewhat extensively in https://github.com/python/typeshed/pull/7773#discussion_r864929231. @janrito said at the time:

> We want to allow str or bytes on actual headers, but None only on updates

Reply:
No, I am not sure. But given that this code does not produce any runtime error:

```Python
session = requests.Session()
session.headers = None
resp = session.get('http://example.com')
assert resp.status_code == 200
```

It appears headers get merged-or-set-to-defaults with None?   https://github.com/psf/requests/blob/main/requests/sessions.py#L61-L71 

I'm new to both typeshed and requests so if it's typeshed's duty to help developers "above and beyond" the runtime code it self I can see how disallowing None could be useful at type checking time.

Reply:
From the mypy_primer output, looks like changing it to `MutableMapping[str, str | bytes] | None` would be quite disruptive -- so I vote for keeping this as `MutableMapping[str, str | bytes]` for now, unless keeping it that way is causing a concrete problem for you.